delegatesAndEventsTask/Task_delegates_events/Task_delegates_events/Program.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task_delegates_events
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            int fileCount = 0;
+            int folderCount = 0;
+            string location = "C:\\Users\\Gopichand_Ulasala\\.NetPractice\\Dummy";
+            List<string> files = new List<string>();
+            files.Add(location + "p2.exe");
+            List<string> dirs = new List<string>();
+            dirs.Add(location + "Keypairs");
+
+            string path = "C:\\Users\\Gopichand_Ulasala\\.NetPractice\\Dummy";
+            FileSystemVisitor fileSystemVisitor = new FileSystemVisitor(path);
+
+            fileSystemVisitor.TraversalStarted += (sender, e) =>
```

Comment:
Refactor all the classes to follow Single Responsibility principle

Reply:
I refactored the code again  and made the entire classes to follow Single Responsibility Principle. Kindly Review it.